visualization/app/codeCharta/ui/searchPanelModeSelector/searchPanelModeSelector.e2e.ts

```diff
@@ -0,0 +1,35 @@
+import { goto, newPage, launch } from "../../../puppeteer.helper"
+import { SearchPanelModeSelectorPageObject } from "./searchPanelModeSelector.po"
+
+describe("SearchPanelModeSelector", () => {
+	let browser, page
+	let searchPanelModeSelector: SearchPanelModeSelectorPageObject
+
+	beforeAll(async () => {
+		browser = await launch()
+	})
+
+	afterAll(async () => {
+		await browser.close()
+	})
+
+	beforeEach(async () => {
+		page = await newPage(browser)
+		searchPanelModeSelector = new SearchPanelModeSelectorPageObject(page)
+
+		await goto(page)
+	})
+
+	it("should open the tree-view when clicking on the tree-view icon", async () => {
+		await searchPanelModeSelector.toggleTreeView()
+
+		expect(await searchPanelModeSelector.isTreeViewOpen()).toBeTruthy()
+	})
+
+	it("should open and close tree-view when clicking on the tree-view icon twice", async () => {
+		await searchPanelModeSelector.toggleTreeView()
```

Comment:
```suggestion
```

It is better to just do both of these things in a single test. The reason is that you would otherwise also have to verify that the first call actually opened it properly. Doing it in the same test also reduces the runtime. These are both favorable things.